### PR TITLE
Allow directives with empty templates

### PR DESCRIPTION
**Pull request by alex94cp**
Related: https://github.com/angular/material/issues/6069

### CODE REVIEW

**Comment by IgorMinar** [comment]
can you please update the commit message to be prefixed `fix(upgrade): .... `

**Comment by IgorMinar** [comment]
can you also add a test please?

**Comment by PascalPrecht** [comment]
This LGTM, can we make it one commit, so that the test is included in the fix?

@alex94puchades can you also rebase?

**Comment by alex94puchades** [comment]
All commits squashed into one. Hope everything is fine now

**Comment by pkozlowski-opensource** [comment]
> All commits squashed into one. Hope everything is fine now

Still needs rebase

**Comment by IgorMinar** [comment]
.... please :-)

**Comment by alex94puchades** [comment]
@pkozlowski-opensource @IgorMinar Haha don't worry, and sorry for the inconvenience!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,51 @@
...
+* End tags used to be tolerated for void elements with no content.
+They are no more allowed so that we more closely follow the HTML5 spec.
+
+* Before
+import * as ng from 'angular2';
+After
+import * as core from 'angular2/core';
+
+
+
```
Comment (IgorMinar) [inline CHANGELOG.md]: this diff should not be here. can you please remove?

**Comment by IgorMinar** [comment]
It looks like this PR is busted. can you please correctly extract your changes and rebase on top of master?

if you don't know how to do this in git, it might be easier to just close this PR and open a new one by copy-pasting the relevant changes to a new branch.

**Comment by alex94puchades** [comment]
Yep, I mistakenly deleted this PR repo so I opened [a new PR](https://github.com/angular/angular/pull/5790)

**Pull request closed without merging**